### PR TITLE
Preliminary implementation of python's threading externs

### PR DESCRIPTION
**Pull request by kevinresol**
https://docs.python.org/3/library/threading.html

### CODE REVIEW

**Comment by kevinresol** [comment]
But I am facing a problem:
```haxe
var thread = new python.lib.threading.Thread({group: null, target: func})
```

when compiled and run, python complains "the argument 'group' should be None"

generated python:
```python
ret = python_lib_threading_Thread(python__KwArgs_KwArgs_Impl_.fromT(_hx_AnonObject({'group': None, 'target': _hx_local_1([tmp])})))
```

File: std/python/lib/threading/Thread.hx [hunk 1]
```diff
@@ -21,6 +21,23 @@
...
+
 @:pythonImport("threading", "Thread")
 extern class Thread {
+  public var name:String;
+  public var ident:Int;
+  public var daemon:Bool;
+  public function new (?options:KwArgs<ThreadOptions>):Void;
+  public function start():Void;
+  public function run():Void;
+  public function join(timeout:Float):Void;
```
Comment (nadako) [inline std/python/lib/threading/Thread.hx]: I think `timeout` argument should be optional.

File: std/python/lib/threading/Lock.hx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+package python.lib.threading;
+
+@:pythonImport("threading", "Lock")
+extern class Lock {
+  public function new():Void;
+  public function acquire(blocking:Bool, timeout:Float):Bool;
```
Comment (nadako) [inline std/python/lib/threading/Lock.hx]: Arguments should be optional, according to the doc.

File: std/python/lib/threading/Thread.hx [hunk 3]
```diff
@@ -21,6 +21,23 @@
  */
 package python.lib.threading;
 
+typedef ThreadOptions = {
+	@:optional var group : Dynamic;
+	@:optional var target : Dynamic;
+	@:optional var name : String;
+	@:optional var args : Tuple<Dynamic>;
+	@:optional var kwargs : KwArgs<Dynamic>;
```
Comment (nadako) [inline std/python/lib/threading/Thread.hx]: Not sure `KwArgs` makes any sense here. Should probably be just `python.Dict<String,Dynamic>`.

**Comment by nadako** [comment]
The group=None thing is actually a compiler bug: https://github.com/HaxeFoundation/haxe/issues/5737.
Other than that, the PR looks good to merge after mentioned stuff fixed.

File: std/python/lib/threading/Thread.hx (hunk 3 above)
Comment (kevinresol) [inline std/python/lib/threading/Thread.hx]: Oh yes, I was mislead by the argument name (`kwargs`)

**Comment by kevinresol** [comment]
Should be fixed now

**Comment by nadako** [comment]
Thanks!

**Pull request merged**